mono/mini/mini-runtime.c

```diff
@@ -1905,6 +1905,97 @@ mono_jit_map_is_enabled (void)
 
 #endif
 
+#if ENABLE_JIT_DUMP
+static FILE* perf_dump_file;
+uint64_t codeIndex = 0;
+pthread_mutex_t mutex = PTHREAD_MUTEX_INITIALIZER;
+void *mmapAddr = MAP_FAILED;
+
+void
+mono_enable_jit_dump (void)
+{
+        if (!perf_dump_file) {
+		char name [64];
+		FileHeader * header = malloc (sizeof (FileHeader));
+		int result = 0;
+		
+		result = pthread_mutex_lock (&mutex);
+		
+		g_snprintf (name, sizeof (name), "/tmp/jit-%d.dump", getpid ());
+		unlink (name);
+		perf_dump_file = fopen (name, "w");
+		
+		add_file_header_info (header);
+		fwrite (header, sizeof (FileHeader), 1, perf_dump_file);
+		
+		mmapAddr = mmap(NULL, sizeof (FileHeader), PROT_READ | PROT_EXEC, MAP_PRIVATE, 1, 0);
+		
+		result = pthread_mutex_unlock (&mutex);
+	}
+}
+
+void
+add_file_header_info (FileHeader *header)
+{
+	header->magic = JIT_DUMP_MAGIC;
```

Comment:
You can initialize `FileHeader` with named initializers when it's allocated in `mono_enable_jit_dump()`; this way, this function can be removed since it's not used anywhere else:

```c
FileHeader header = {
   .magic = JIT_DUMP_MAGIC,
   .version = JIT_DUMP_VERSION,
   ...
};
```

Reply:
Please do not use designated initializers.
The runtime is all valid C++ and this would not be.